### DISCUSSION TITLE
Java RAM Verbrauch begrenzen

### DISCUSSION THREAD

**Discussion by zeron850** [camueller/SmartApplianceEnabler#335] [Installation und Betrieb]
Hallo Alex,

ich nutze dein SAE gemeinsam mit ioBroker auf einem Odroid C4 mit 3,7 GB RAM. Wenn der C4 RAM langsam aufgebraucht wird, killt mein OS (Dietpi 8.8.1) irgendwann den Prozess der am meisten RAM, in dem Fall SAE 2.0.5 Beta. Gibt es eine Möglichkeit den RAM Verbrauch des SAE zu limitieren?

Danke

**Comment by camueller** [comment]
Der Java-Heap ist beim [SAE auf 256 MB begrenzt](https://github.com/camueller/SmartApplianceEnabler/blob/master/run/etc/default/smartapplianceenabler). 
Ich konnte bisher auch keine Probleme diesbezüglioch beobachten. Möglicherweise liegt das an der Java-Version, die Du auf dem Odroid verwendest. Diese Platform wird vom SAE nicht offiziell unterstützt.

**Reply by zeron850** [reply]
Danke für deine schnelle Antwort. Stellenweise geht der SAE auf über 700 MB. Ich verwende als Java Version Javajdk 17.0.4

Ich habe jetzt mal die Java Version 11 installiert und versuche ob damit der Memory Leak verschwindet.

**Reply by zeron850** [reply]
Ich habe jetzt Java Version 11 installiert und er benutzt weit über 600 MB. Hast du mir evtl einen Tipp an was es sonst noch liegen kann?

**Reply by camueller** [reply]
Habe gerade mal geschaut. Bei mir verwendet der SAE aktuell 570 MB, wobei ich ihn vor einer Woche zwecks Testen einer neuen Version neu gestartet habe. Aber die absolute Zahl ist eigentlich egal - jedenfalls hatte ich noch nie Probleme wegen Speichermangel oder dass der Speicher vollläuft. Einer meiner produktiven Raspis läuft mit Docker seit Monaten.

**Reply by zeron850** [reply]
Die 570 MB oder meine 750 MB wäre dann kein Problem wenn es die einzige Software auf dem Pi etc. wäre. Bei mir kommt noch ioBroker dazu welcher einen sehr hohen Ressourcenbedarf hat und irgendwann killt das OS dann den Prozess mit dem höchsten Bedarf wenn der RAM voll ist. Deshalb wäre es gut wenn der SAE eben nicht mehr als 300-400 MB in Summe verbrauchen würde. Der SAE steigt halt von Stunde zu Stunde an und gönnt sich immer mehr RAM

Demnach funktioniert die maxHeapSize bei dir aber auch nicht?

**Reply by camueller** [reply]
Der Heap ist ja nur ein Teil des Speichers, den der Prozess braucht.

Bei mir steigt der Speicherverbraucht nicht.

**Reply by zeron850** [reply]
Ok gibt es eine Möglichkeit den Speicherbedarf neben dem Heap zu begrenzen bzw. einzufrieren?

**Reply by camueller** [reply]
Nochmal: auf dem Raspberry Pi gibt es kein Problem. Probleme auf anderen Platformen oder mit anderen Java-Versionen musst Du selbst lösen.
